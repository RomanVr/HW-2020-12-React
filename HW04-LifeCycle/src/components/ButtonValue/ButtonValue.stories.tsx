import React from "react";

import { ButtonValue, ButtonValueProps } from "./ButtonValue";
import { Meta, Story } from "@storybook/react";

export default {
  title: "Form/ButtonValue",
  component: ButtonValue,
} as Meta;

const Template: Story<ButtonValueProps> = (args) => <ButtonValue {...args} />;

export const ButtonStory = Template.bind({});

ButtonStory.args = {
  value: "Show Time",
};
