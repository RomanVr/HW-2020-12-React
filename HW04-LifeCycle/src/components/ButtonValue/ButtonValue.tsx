import React from "react";

interface ButtonValueProp {
  value: string;
  handleClick: () => void;
}
export const ButtonValue: React.FC<ButtonValueProp> = ({
  value,
  handleClick,
}) => <input type="button" value={value} onClick={handleClick} />;
