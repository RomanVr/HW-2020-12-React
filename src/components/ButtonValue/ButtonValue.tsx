import React from "react";

export interface ButtonValueProps {
  value: string;
  handleClick: () => void;
}
export const ButtonValue: React.FC<ButtonValueProps> = ({
  value,
  handleClick,
}) => (
  <input
    data-testid="buttonValue"
    type="button"
    value={value}
    onClick={handleClick}
    style={{ display: "inline-block" }}
  />
);
