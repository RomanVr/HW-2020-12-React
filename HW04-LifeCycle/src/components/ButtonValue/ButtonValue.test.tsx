import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { ButtonValue } from "./ButtonValue";

describe("ButtonValue", () => {
  it("Render button for empty value", () => {
    expect(
      renderer.create(<ButtonValue value="" handleClick={jest.fn()} />).toJSON()
    ).toMatchSnapshot();
  });

  it("Render button for filled with 'Show time'", () => {
    expect(
      renderer
        .create(<ButtonValue value="Show time" handleClick={jest.fn()} />)
        .toJSON()
    ).toMatchSnapshot();
  });

  it("calls handleClick callback on click by empty button", () => {
    const handleClick = jest.fn();
    const wrapper = mount(<ButtonValue value="" handleClick={handleClick} />);
    wrapper.simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });
});
