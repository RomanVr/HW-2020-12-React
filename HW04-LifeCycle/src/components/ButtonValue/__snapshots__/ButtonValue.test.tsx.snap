// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonValue Render button for empty value 1`] = `
<input
  data-testid="buttonValue"
  onClick={[MockFunction]}
  type="button"
  value=""
/>
`;

exports[`ButtonValue Render button for filled with 'Show time' 1`] = `
<input
  data-testid="buttonValue"
  onClick={[MockFunction]}
  type="button"
  value="Show time"
/>
`;
